test/test_sparse.py

```diff
@@ -105,6 +105,9 @@ def sparse_tensor_factory(*args, **kwargs):
         self.legacy_sparse_tensor = torch.sparse.DoubleTensor
 
     def _gen_sparse(self, sparse_dim, nnz, with_size, dtype, device, coalesced):
+        if nnz < 2:
+            assert coalesced, "Tensor with nnz < 2 is always coalesced"
+
```

Comment:
This is not always true. For instance, when creating a sparse COO tensor with `nnz<2`, one can always use `_coalesced_(False)` method call to make it uncoalesced:
```python
>>> a = torch.tensor([]).to_sparse()
>>> a.is_coalesced()
True
>>> b=a._coalesced_(False);
>>> a.is_coalesced()
False
```
In fact, this property allows to keep the current sparse coo tests mostly untouched if genSparseTensor implementation
https://github.com/pytorch/pytorch/blob/806878518f32c5b93acc7da576e57ab52f6f5232/torch/testing/_internal/common_utils.py#L2270-L2279
is updated as follows:
```python
        if not is_uncoalesced:
            x = x.coalesce()
        else:
            # ...
            x = x.detach().clone()._coalesced_(False)
```
and here have
```suggestion
```